src/histolab/tiler.py

```diff
@@ -77,20 +77,48 @@ def __init__(
 
         super().__init__()
 
-        assert (
-            max_iter >= n_tiles
-        ), "The maximum number of iterations must be grater than or equal to the "
-        f"maximum number of tiles. Got max_iter={max_iter} and n_tiles={n_tiles}."
-
         self.tile_size = tile_size
+        self.n_tiles = n_tiles
         self.max_iter = max_iter
         self.level = level
-        self.n_tiles = n_tiles
         self.seed = seed
         self.check_tissue = check_tissue
         self.prefix = prefix
         self.suffix = suffix
 
+    @property
+    def tile_size(self) -> Tuple[int, int]:
+        return self._valid_tile_size
+
+    @tile_size.setter
+    def tile_size(self, tile_size_: Tuple[int, int]):
+        if tile_size_[0] < 1 or tile_size_[1] < 1:
```

Comment:
note: 0 is considered positive, so or this condition is wrong or the error message is flaky :D